packages/@aws-cdk/aws-pipes-alpha/lib/enrichment.ts

```diff
@@ -0,0 +1,40 @@
+import { IRole } from 'aws-cdk-lib/aws-iam';
+import { CfnPipe } from 'aws-cdk-lib/aws-pipes';
+import { IPipe } from '.';
+
+/**
+ * The parameters required to set up enrichment on your pipe.
+ *
+ * @see http://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-properties-pipes-pipe-pipeenrichmentparameters.html
+ */
+export interface EnrichmentParametersConfig {
+
+  /**
+   * The parameters for the enrichment target.
+   */
+  readonly enrichmentParameters: CfnPipe.PipeEnrichmentParametersProperty;
+
+  /**
+   * The ARN of the enrichment resource.
+   *
+   * Length Constraints: Minimum length of 0. Maximum length of 1600.
+   * @see https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-resource-pipes-pipe.html#cfn-pipes-pipe-enrichment
+   */
+  readonly enrichmentArn: string;
+}
+
+/**
+ * Enrichment step to enhance the data from the source before sending it to the target.
+ */
+export interface IEnrichment {
+  /**
+   * Grant the pipes role to invoke the enrichment.
+   */
+  grantInvoke(grantee: IRole): void;
+
+  /**
+   * Bind this enrichment to a pipe.
+   */
+  bind(pipe: IPipe): EnrichmentParametersConfig;
```

Comment:
```suggestion
  bind(pipe: IPipe): CfnPipe.PipeEnrichmentParametersProperty;
```

Reply:
I am not sure if that's what we want? The cfn interface is just the parameters for the enrichment. But besides the parameters we need a ARN to the enrichment. 
The interface `EnrichmentParametersConfig` has to attributes. The parameters which are referenced by `CfnPipe.PipeEnrichmentParametersProperty` and an attribute for the enrichment ARN. 
If the bind should just return the parameters, where should the ARN be defined?

Reply:
Isn't that what `IEnchrichment` is for?

In my head the definition of `bind` is strictly "convert this L2 to the respective properties for the L1`>

Reply:
Ok, I will change that.

Reply:
Makes sense. Let me look into this rule.

Reply:
Yes. Alright then. 😂